src/state.rs

```diff
@@ -193,3 +196,18 @@ pub(crate) struct StateSnapshot {
     transaction_number: u16,
     context_u128: u128,
 }
+
+#[derive(Debug, Clone, Default)]
+pub struct CycleCounts {
+    pub keccak256_cycles: usize,
+    pub ecrecover_cycles: usize,
+    pub sha256_cycles: usize,
+    pub secp256v1_verify_cycles: usize,
+    pub code_decommitter_cycles: usize,
+}
+
+impl PartialEq for CycleCounts {
```

Comment:
I'd argue that this is an extremely weird impl. It would be better to exclude `cycle_counts` field from the `PartialEq` impl for `State` given that after my PR, there's a manual impl already.